File: crates/chidori-js/benchmarks/workloads/array_sum.js
// Dense-array traversal with index arithmetic — the `s += a[i]` class the
// typed loop kernels' element access targets: reads, in-place writes, a
// two-array dot product, and a nested 2D walk. Deterministic fill (no RNG)
// so every runtime computes the same checksum.
const N = 200_000;
const ROUNDS = 5;
const a = new Array(N);
const b = new Array(N);
for (let i = 0; i < N; i++) {
  a[i] = (i * 7919) % 10007;
  b[i] = (i * 104729) % 7919;
}
let checksum = 0;
for (let r = 0; r < ROUNDS; r++) {
  // read + accumulate
  let s = 0;
  for (let i = 0; i < a.length; i++) {
    s += a[i];
  }
  // dot product
  let d = 0;
  for (let i = 0; i < a.length; i++) {
    d += a[i] * b[i];
  }
  // in-place transform
  for (let i = 0; i < a.length; i++) {
    a[i] = (a[i] + b[i]) % 10007;
  }
  checksum = (checksum + s + d) % 9007199254740991;
}
// nested 2D walk
let m = 0;
for (let i = 0; i < 500; i++) {
  for (let j = 0; j < 500; j++) {
    m += (i * j) % 13;
  }
}
console.log("RESULT=" + (checksum + m));

File: crates/chidori-js/benchmarks/workloads/array_sum.py
# Python twin of array_sum.js — dense-list traversal with index arithmetic:
# reads, in-place writes, a two-list dot product, and a nested 2D walk.
# Deterministic fill (no RNG) so every runtime computes the same checksum.
#
# The checksum is accumulated in float, not int: JS does this math in IEEE
# doubles, and `checksum + s + d` can graze 2^53 where doubles round. Python
# floats reproduce that rounding exactly; exact ints would not.
N = 200_000
ROUNDS = 5
a = [0] * N
b = [0] * N
for i in range(N):
    a[i] = (i * 7919) % 10007
    b[i] = (i * 104729) % 7919
checksum = 0.0
for r in range(ROUNDS):
    # read + accumulate
    s = 0
    for i in range(len(a)):
        s += a[i]
    # dot product
    d = 0
    for i in range(len(a)):
        d += a[i] * b[i]
    # in-place transform
    for i in range(len(a)):
        a[i] = (a[i] + b[i]) % 10007
    checksum = (checksum + s + d) % 9007199254740991
# nested 2D walk
m = 0
for i in range(500):
    for j in range(500):
        m += (i * j) % 13
print("RESULT=" + str(int(checksum + m)))
